src/Castle.Windsor.Tests/TypeUtilTestCase.cs

```diff
@@ -55,38 +55,38 @@ public void Closed_generic_simple_type()
 		}
 
 		[Test]
-		public void Generic_nested_generic_typeArray_multi_dimentional_pulls_closed_generics_to_innermost_type()
+		public void Generic_nested_generic_typeArray_multi_dimensional()
 		{
 			var name = typeof(GenericHasNested<A2>.NestedGeneric<AProp>[,,]).ToCSharpString();
-			Assert.AreEqual("GenericHasNested<�TOuter�>.NestedGeneric<A2, AProp>[,,]", name);
+			Assert.AreEqual("GenericHasNested<A2>.NestedGeneric<AProp>[,,]", name);
 		}
 
 		[Test]
-		public void Generic_nested_generic_typeArray_pulls_closed_generics_to_innermost_type()
+		public void Generic_nested_generic_typeArray()
 		{
 			var name = typeof(GenericHasNested<A2>.NestedGeneric<AProp>[]).ToCSharpString();
-			Assert.AreEqual("GenericHasNested<�TOuter�>.NestedGeneric<A2, AProp>[]", name);
+			Assert.AreEqual("GenericHasNested<A2>.NestedGeneric<AProp>[]", name);
 		}
 
 		[Test]
-		public void Generic_nested_generic_type_pulls_closed_generics_to_innermost_type()
+		public void Generic_nested_generic_type()
 		{
 			var name = typeof(GenericHasNested<A2>.NestedGeneric<AProp>).ToCSharpString();
-			Assert.AreEqual("GenericHasNested<�TOuter�>.NestedGeneric<A2, AProp>", name);
+			Assert.AreEqual("GenericHasNested<A2>.NestedGeneric<AProp>", name);
 		}
 
 		[Test]
-		public void Generic_nested_type_array_ignores_outer_generic_argument()
+		public void Generic_nested_type_array()
 		{
 			var name = typeof(GenericHasNested<A2>.Nested[]).ToCSharpString();
-			Assert.AreEqual("GenericHasNested<�TOuter�>.Nested<A2>[]", name);
+			Assert.AreEqual("GenericHasNested<A2>.Nested[]", name);
 		}
 
 		[Test]
-		public void Generic_nested_type_ignores_outer_generic_argument()
+		public void Generic_nested_type()
 		{
 			var name = typeof(GenericHasNested<A2>.Nested).ToCSharpString();
-			Assert.AreEqual("GenericHasNested<�TOuter�>.Nested<A2>", name);
+			Assert.AreEqual("GenericHasNested<A2>.Nested", name);
```

Comment:
Should we add a test that nests generic arguments? 

```csharp
var name = typeof(GenericHasNested<A2>.GenericNested<Tuple<int, bool>>).ToCSharpString();
Assert.AreEqual("GenericHasNested<A2>.GenericNested<Tuple<int, bool>>", name);
```

Reply:
Will do. It's good to have a test to see whether the recursive type forwarding happens only where it should.

Beyond that, I would caution against blowing up this utility part of Windsor too much. If we wanted a really robust implementation for type name formatting, I'd be happy to donate my own recent work (stalx/TypeNameFormatter) which comes tested and as a source code NuGet package that easily embeddable (in the spirit of not reinventing the wheel and doing double work). That said, I perfectly understand if Castle would rather stay free of 3rd-party code.

Reply:
I did have a close look at that, it is mental the acrobatics one has to do to achieve this. I would like to avoid a third party dependency on this but ultimately it is for @jonorossi to decide. Thanks for doing this. The last test case you added was where I got up to :)

Reply:
I've all done it before so it wasn't too bad this rime around.

Try a type such as `int[][,][,,]` and see what happens to the commas. ;-)

Totally fine with *not* introducing a dependency, I just thought I'd mention it as a possibility.

Reply:
Thanks for the offer @stakx, I don't mind the contribution though would prefer it to be embedded rather than fetched from NuGet, to be embedded it probably would also be best licensed under Apache 2. However I think what we've now got covers what the debugger views are going to have to worry about anyway, I can't imagine any user of Windsor having a component of type `int[][,][,,]` (can't imagine Windsor resolving it 😉), so for pragmatism sake I think your fix here is all we need.

Reply:
>  I can't imagine any user of Windsor having a component of type int[][,][,,]

You'd be surprised what contraptions people come up with when you least expect it! :laugh:

> so for pragmatism sake I think your fix here is all we need.

Agreed, let's just wait and see if anyone complains about this again.

Reply:
> You'd be surprised what contraptions people come up with when you least expect it! 😆

Duuuude. I feel you on that.